### ISSUE TITLE
do-template linter should warn if the values given to it don't make sense

### ISSUE THREAD

**Issue by imrekoszo** [clj-kondo/clj-kondo#2475]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

do-template is understood by clj-kondo already. It would be very handy to have findings registered when
1. there were no values supplied to do-template
2. number of values supplied is not a multiple of the number of args in the arg vector

Justification:

[do-template](https://clojuredocs.org/clojure.template/do-template)'s docstring says:

```
Repeatedly copies expr (in a do block) for each group of arguments
in values.  values are automatically partitioned by the number of
arguments in argv, an argument vector as in defn.
 Example: (macroexpand '(do-template [x y] (+ y x) 2 4 3 5))
         ;=> (do (+ 4 2) (+ 5 3))
```

If you happen to misplace a paren, do-template might not generate any code at all:

```clojure
(macroexpand
 '(do-template
   [a b]
   (let [result (+ a b)]
     result

     1 2

     3 4))) ;; misplaced closing paren of let
=> (do)
```

If your value count isn't divisible with argcount, trailing values are skipped:

```clojure
(macroexpand
 '(do-template
   [a b]
    (let [result (+ a b)]
      result)

    1 2

    3 
    ))
=> (do (let [result (+ 1 2)] result))
```

**Describe the solution you'd like**
Findings to be registered when do-template has either 0 values or (count values) not divisible by (count args).

**Describe alternatives you've considered**
I considered adding an analyze-call hook, ~but if I do this I don't get the built-in unwrapping of the do-template~. Edit: if the hook returns `nil` then we do get the built-in unwrapping, see the first comment below.

I also tried analyze-call plus macroexpanding do-template with itself, but these two don't work together sadly.

**Additional context**
I believe the findings could be emitted around here: https://github.com/clj-kondo/clj-kondo/blob/1367dadd93b36e850b8eeebc5346cd99446063ed/src/clj_kondo/impl/macroexpand.clj#L196-L197

If the idea is welcome I can try putting together a PR.

**Comment by borkdude** [comment]
> I considered adding an analyze-call hook, but if I do this I don't get the built-in unwrapping of the do-template.

You do if you return `nil` from the result of the hook

**Comment by borkdude** [comment]
> I considered adding an analyze-call hook, but if I do this I don't get the built-in unwrapping of the do-template.

You do if you return `nil` from the result of the hook

**Comment by imrekoszo** [comment]
> You do if you return nil from the result of the hook

Cheers, TIL. I must have overlooked this in the documentation.

**Comment by imrekoszo** [comment]
Proposed a fix in https://github.com/clj-kondo/clj-kondo/pull/2476/

**Comment by imrekoszo** [comment]
I haven't added a default config yet as I don't know whether this linter is welcome. Suggestion would be warning.

**Comment by imrekoszo** [comment]
> test a couple of projects

Makes sense. Do you have a tried way of doing that?
